src/client/datascience/variablesView/variableView.ts

```diff
@@ -215,4 +216,22 @@ export class VariableView extends WebviewViewHost<IVariableViewPanelMapping> imp
     private async activeNotebookRestarted() {
         this.postMessage(InteractiveWindowMessages.RestartKernel).ignoreErrors();
     }
+
+    private async sendRefreshMessage() {
+        if (this.notebookWatcher.activeNotebookExecutionCount !== undefined) {
+            this.postMessage(InteractiveWindowMessages.UpdateVariableViewExecutionCount, {
+                executionCount: this.notebookWatcher.activeNotebookExecutionCount
+            }).ignoreErrors();
+
+            const args: IJupyterVariablesRequest = {
+                executionCount: this.notebookWatcher.activeNotebookExecutionCount,
+                sortColumn: 'name',
+                startIndex: 0,
+                sortAscending: true,
+                pageSize: 5,
+                refreshCount: 1
+            };
+            await this.requestVariables(args);
```

Comment:
Weird
* this.variables.refreshRequired(this.sendRefreshMessage, this, this.disposables);
	So, lets assume `variables` object triggers the event `refreshRequired`
* The handler will then do something and call a method `requestVariables`
* The `requestVariables` method then calls a method in `variables` object

I.e. its just a loop, variables fires & event indicating variables need to be refreshed & then something else calls a method on that same object.

Or is the event name incorrect? Does the event mean the variables are stale?

@rchiodo @DavidKutu  not sure you can help here, but @IanMatthewHuff  is out.
If not sure, then we can ignore this. but the flow seems to be weird

Reply:
I agree, this seems like it could be refactored. The responsibilities of the different classes are getting muddied in my head.

Reply:
this was also removed